cmd/event_display/main.go

```diff
@@ -47,15 +49,42 @@ Data,
   }
 */
 
+// display prints the given Event in a human-readable format.
 func display(event cloudevents.Event) {
-	fmt.Printf("☁️  cloudevents.Event\n%s", event.String())
+	fmt.Printf("☁️  cloudevents.Event\n%s", event)
 }
 
 func main() {
-	c, err := cloudevents.NewClientHTTP()
+	run(context.Background())
+}
+
+func run(ctx context.Context) {
+	c, err := cloudevents.NewClientHTTP(
+		cehttp.WithMiddleware(healthzMiddleware()),
+	)
```

Comment:
@n3wscott I needed something quick so I used a `Middleware` for the time being, but this feels clumsy to me (see the implementation of `healthzMiddleware()` below which is especially hacky).

Since I'm going to refactor this for testability, I thought now was a good time to ask for your expertise :)